libs/auth-api-lib/src/lib/services/user-profile.service.ts

```diff
@@ -119,47 +144,58 @@ export class UserProfileService {
   }
 
   private formatAddress(
-    address: NationalRegistryAddress | null,
+    address: Address | null,
+    hasCountryCode = false,
   ): AddressDTO | undefined {
     if (!address) {
       return undefined
     }
 
+    // When individuals live abroad, the national registry stores the country name under "heiti" or streetAddress.
     const likelyForeign =
       address.streetAddress != null &&
       address.locality == null &&
       address.postalCode == null &&
       address.municipalityNumber == null
     if (likelyForeign) {
-      // When individuals live abroad, the national registry stores the country name under "heiti".
       return {
         country: address.streetAddress,
       }
     }
 
+    const country = this.formatCountry(address.country, hasCountryCode)
     return {
-      formatted: this.formattedAddressString(address),
+      formatted: this.formattedAddressString(address, country),
       streetAddress: address.streetAddress ?? undefined,
       locality: address.locality ?? undefined,
       postalCode: address.postalCode ?? undefined,
+      country,
```

Comment:
Everything can be `undefined` except country which will be `Ísland` if undefined. I'm wondering if we could get some weird edge cases by providing default value for country, and if we should instead allow country to be undefined?

Reply:
That makes sense.